### PR TITLE
Align table query parameters with Trace Server Protocol

### PR DESCRIPTION
**Pull request by PatrickTasse**
Signed-off-by: Patrick Tasse <patrick.tasse@gmail.com>

### CODE REVIEW

**Comment by PatrickTasse** [comment]
I didn't notice that the parameter keys were specified in the body text paragraph in https://theia-ide.github.io/trace-server-protocol/#operation/getLines. I was looking at the 'parameters' description below and the 'Payload' example.

The keys are listed somewhere else in the source TSP but it doesn't show in this web version.

Nevertheless I would suggest to change also the TSP documentation to align with these keys. We should also find a way to update the documentation so that each different query can have its own description of the parameters that apply to it.

**Comment by bhufmann** [comment]
> I didn't notice that the parameter keys were specified in the body text paragraph in https://theia-ide.github.io/trace-server-protocol/#operation/getLines. I was looking at the 'parameters' description below and the 'Payload' example.
> 
> The keys are listed somewhere else in the source TSP but it doesn't show in this web version.
> 
> Nevertheless I would suggest to change also the TSP documentation to align with these keys. We should also find a way to update the documentation so that each different query can have its own description of the parameters that apply to it.

These keys should be defined in the TSP and the implementation here needs to be aligned. I see that the web viewer of the TSP is not best when it comes to formatting as well as showing the schema instead of the example. 

Maybe update the title of the PR since we don't want to align to Trace Compass internal implementation. Maybe something along the lines " ... align with TSP ..." or make "... make keys consistent with other keys ...".

In any case, I agree the TSP documentation needs to be updated.

File: src/models/query/query-helper.ts [hunk 1]
```diff
@@ -17,17 +17,17 @@ export class QueryHelper {
...
      * Key for the number of element to return
      */
-    public static readonly REQUESTED_TABLE_COUNT_KEY: string = 'size';
+    public static readonly REQUESTED_TABLE_COUNT_KEY: string = 'requested_table_count';
 
     /**
      * Table column IDs key
      */
-    public static readonly REQUESTED_COLUMN_IDS_KEY = 'columnIds';
+    public static readonly REQUESTED_COLUMN_IDS_KEY = 'requested_table_column_ids';
```
Comment (MatthewKhouzam) [inline src/models/query/query-helper.ts:30]: Should the constant be "REQUESTED_TABLE_COLUMN_IDS_KEY"

**Comment by tahini** [comment]
> 
> These keys should be defined in the TSP and the implementation here needs to be aligned. I see that the web viewer of the TSP is not best when it comes to formatting as well as showing the schema instead of the example.
> 

+1 This should align with the TSP and the TSP should not align with whatever naming we gave in TC, but make sense from a protocol implementer/consumer point of view. TSP may need to be updated first.

**Review by MatthewKhouzam — CHANGES_REQUESTED** [review]
The more I think of it, the more I think we shouldn't change the TSP to be tracecompass dependant.

If we improve the tsp that's fine, but we need to be agnostic first.

**Comment by PatrickTasse** [comment]
TSP is currently inconsistent, here is where the alternative keys are listed:

https://github.com/theia-ide/trace-server-protocol/blob/master/API.yaml#L1904

However the "_key" suffixes are wrong there...

**Comment by tahini** [comment]
FTR, I did a "TSP alignment" phase recently (see https://github.com/theia-ide/trace-server-protocol/pull/27), relating to model response data (I did not really touch the query parameters  then). My un-discussed and maybe dictatorial rule of thumb was to name to fields in camelCase and remove superfluous keywords (it's data being passed after all), so 'startTime' becomes simply 'start'.

In the trace server, I just wrote custom serializers to change TC field names into the TSP fields. No change required to TC code or API changes.

Something similar can be done for the requests. It would be normal that a server implementing a protocol needs to map the protocol's parameters to its internal equivalent of those.

**Comment by PatrickTasse** [comment]
It was pointed out to me that according to this patch https://git.eclipse.org/r/143076, the parameter keys were meant to be updated in the TSP to the new ones I use in this commit. I guess the TSP update was not complete.

It has been suggested to update the TSP documentation first, then align to the new TSP parameter keys with this commit. I'll make a commit on the TSP.

**Review by MatthewKhouzam — APPROVED** [review]
Now it works!

**Pull request merged**